### PR TITLE
[ new ] Bitwise XOR for Bits64 and Integer

### PR DESCRIPTION
**Pull request by stefan-hoeck**
This adds primitives `prim__xor_Bits64` and `prim__xor_Integer` to the compiler as discussed in #983. In addition, tests for all primitive bit ops have been added (which led to the detection of  #1023   :-) ).

I'll submit a new PR for including these in `bxor` in `src/Core/Primitives.idr` once the new compiler is released.

### CODE REVIEW

**Comment by gallais** [comment]
You probably need to move your `basic054` test case because #1023 snatched that name.

**Comment by stefan-hoeck** [comment]
> You probably need to move your basic054 test case because #1023 snatched that name.

Hmm, is this still an issue? I merge master into this branch and resolved conflicts (at least I thought so) after #1023 was committed. (See d84a5e8)

**Comment by gallais** [comment]
Oh, I see. It thought you had overwritten `basic054` during your merge.

File: tests/idris2/basic054/input [hunk 1]
```diff
@@ -0,0 +1,2 @@
+:exec main
+:q
```
Comment (gallais) [inline tests/idris2/basic054/input:2]: Note that this file is useless: `run` does not use an `input`.

Comment (stefan-hoeck) [reply tests/idris2/basic054/input]: OK. Removed the file.

**Comment by stefan-hoeck** [comment]
> Edit: there are indeed changes though, coming from a broken merge rather
> than pre-emptively moving the test before merging AFAICT.

There is now only `tests/idris2/basic054/run`, which seems to be identical to your version. Why it still appears in the list of changed files, I don't know. Should I try and do something about that?

Also, some tests are failing. Are these failures to be expected here?

**Comment by stefan-hoeck** [comment]
I just saw that I forget the '--cg node' option in the node tests, so please do not merge this yet. Will look into it tomorrow.

**Comment by stefan-hoeck** [comment]
The node tests fail in several ways (for instance #1037 and #1039). I'll therefore do the following: I'll try and fix those issues first, adding most of the bitwise tests in this PR to the PRs with the fixes. Once the fixes are merged, I'll rebase and continue here.

**Comment by stefan-hoeck** [comment]
I rebased this and activated the additional xor-tests. This should be ready for reviewing now.

**Pull request merged**